design-system/pkg/src/number-field/NumberField.tsx

```diff
@@ -50,23 +58,25 @@ export const NumberField: ForwardRefExoticComponent<
 
   return (
     <TextFieldPrimitive
+      ref={inputRef}
```

Comment:
Ref wasn't passed on properly + add missing props `contextualHelp`, and `isRequired`.